File: java/backend/src/main/java/com/example/entity/Cartitem.java
package com.example.entity;

import jakarta.persistence.*;

import java.math.BigDecimal;

@Entity
@Table(name = "cartitem")
public class Cartitem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "CartItem_Id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "Cart_Id", nullable = false)
    private Cart cart;

  
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "Prod_Id", nullable = false)
    private Product prod;

    @Column(name = "Quantity", nullable = false)
    private Integer quantity;

    @Column(name = "PriceSnapshot", nullable = false, precision = 10, scale = 2)
    private BigDecimal priceSnapshot;

   
    @Column(name = "price_type", length = 10)
    private String priceType = "MRP";

  
    @Column(name = "points_used")
    private Integer pointsUsed = 0;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Cart getCart() {
        return cart;
    }

    public void setCart(Cart cart) {
        this.cart = cart;
    }

    public Product getProd() {
        return prod;
    }

    public void setProd(Product prod) {
        this.prod = prod;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public BigDecimal getPriceSnapshot() {
        return priceSnapshot;
    }

    public void setPriceSnapshot(BigDecimal priceSnapshot) {
        this.priceSnapshot = priceSnapshot;
    }

    public String getPriceType() {
        return priceType;
    }

    public void setPriceType(String priceType) {
        this.priceType = priceType;
    }

    public Integer getPointsUsed() {
        return pointsUsed;
    }

    public void setPointsUsed(Integer pointsUsed) {
        this.pointsUsed = pointsUsed;
    }

}

File: dotnet/backend/Models/Cartitem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace EMart.Models
{
    [Table("cartitem")]
    public class Cartitem
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("cart_item_id")]
        public int Id { get; set; }

        [Required]
        [Column("cart_id")]
        public int CartId { get; set; }

        [ForeignKey("CartId")]
        [JsonIgnore]
        public virtual Cart Cart { get; set; } = null!;

        [Required]
        [Column("prod_id")]
        public int ProductId { get; set; }

        [ForeignKey("ProductId")]
        public virtual Product Product { get; set; } = null!;

        [Required]
        [Column("quantity")]
        public int Quantity { get; set; }

        [Required]
        [Column("price_snapshot", TypeName = "decimal(10,2)")]
        public decimal PriceSnapshot { get; set; }

       
        [Column("price_type")]
        [MaxLength(10)]
        public string PriceType { get; set; } = "MRP";

       
        [Column("points_used")]
        public int PointsUsed { get; set; } = 0;
    }
}
